### ISSUE TITLE
watchtower: add a configurable retry on RPC timeouts

### ISSUE THREAD

**Issue by mvines** [solana-labs/solana#9884]
Occasional timeouts on RPC operations appear to be triggering false error alerts.  One likely example:
```
[2020-05-05T15:50:29.812757337Z INFO  solana_metrics::metrics] datapoint: watchtower-sanity ok=true
[2020-05-05T15:51:49.812958232Z INFO  solana_client::rpc_client_request] GetTransactionCount failed, 0 retries left: reqwest::Error { kind: Request, url: "http://tds.solana.com/", source: TimedOut }
[2020-05-05T15:51:49.813026802Z INFO  solana_metrics::metrics] datapoint: watchtower-sanity ok=false
[2020-05-05T15:51:50.056658888Z ERROR solana_metrics::metrics] datapoint: watchtower-sanity-failure test="rpc" err="error sending request for url (http://tds.solana.com/): operation timed out"
[2020-05-05T15:52:51.047243311Z INFO  solana_watchtower] Current transaction count: 491688893
[2020-05-05T15:52:51.047267266Z INFO  solana_watchtower] Recent blockhash: 6DDnVm3hEedmuJnPXMbpYFYrvMUmz8PVCEjmBjkB7izX
```

Rather than raising the alarm at the first RPC timeout, it would be nice to retry a couple times (perhaps make this a command-line option) to avoid walking people up unnecessarily

**Comment by bunghi** [comment]
+1

I receive a lot of rpc timeout messages from watchtower

**Comment by mvines** [comment]
Ah, turns out I fixed this issue recently in v1.4 with this argument via https://github.com/solana-labs/solana/pull/13846:
```
--unhealthy-threshold <COUNT>
            How many consecutive failures must occur to trigger a notification [default: 1]
```

**Comment by bunghi** [comment]
https://github.com/solana-labs/solana/issues/13499 can be merged
